library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt

```diff
@@ -0,0 +1,400 @@
+
+    private var currentPage: Int = 0
+
+    private var calendarProperties: CalendarProperties = CalendarProperties(context)
+    private var calendarPageAdapter: CalendarPageAdapter = CalendarPageAdapter(context, calendarProperties)
+
+    private val onPageChangeListener = object : ViewPager.OnPageChangeListener {
+        /**
+         * This method set calendar header label
+         *
+         * @param position Current ViewPager position
+         * @see ViewPager.OnPageChangeListener
+         */
+        override fun onPageSelected(position: Int) {
+            val calendar = calendarProperties.firstPageCalendarDate?.clone() as Calendar
+            calendar.add(Calendar.MONTH, position)
+
+            if (!isScrollingLimited(calendar, position)) {
+                setHeaderName(calendar, position)
+            }
+        }
+
+        override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
+        override fun onPageScrollStateChanged(state: Int) {}
+    }
+
+    /**
+     * @return List of Calendar object representing a selected dates
+     */
+    var selectedDates: List<Calendar>
+        get() = calendarPageAdapter.selectedDays
+                .mapNotNull { it.calendar }
+        set(selectedDates) {
+            calendarProperties.selectDays(selectedDates)
+            calendarPageAdapter.notifyDataSetChanged()
+        }
+
+    /**
+     * @return Calendar object representing a selected date
+     */
+    val selectedDate: Calendar?
+        get() = calendarPageAdapter.selectedDays.map { it.calendar }.first()
+
+    /**
+     * @return Calendar object representing a date of current calendar page
+     */
+    private val currentPageDate: Calendar
+        get() {
+            val calendar = calendarProperties.firstPageCalendarDate?.clone() as Calendar
+            return calendar.apply {
+                set(Calendar.DAY_OF_MONTH, 1)
+                add(Calendar.MONTH, calendarViewPager.currentItem)
+            }
+        }
+
+    init {
+        properties?.let { this.calendarProperties = properties }
+        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        inflater.inflate(R.layout.calendar_view, this)
+        initControl(context, attrs)
+        initAttributes()
+        initCalendar()
+    }
+
+    private fun initControl(context: Context, attrs: AttributeSet?) {
+        calendarProperties = CalendarProperties(context)
+
+        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        inflater.inflate(R.layout.calendar_view, this)
+
+        initUiElements()
+        attrs?.let { setAttributes(attrs) }
+    }
+
+    /**
+     * This method set xml values for calendar elements
+     *
+     * @param attrs A set of xml attributes
+     */
+    private fun setAttributes(attrs: AttributeSet) {
+        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.CalendarView)
```

Comment:
Tą całą metodę można chyba zastąpić:
context.withStyledAttributes(attrs, R.styleable.CalendarView, defStyleAttr, 0) {
initCalendarProperties(this)
initAttributes()
}